cmd/registry/cmd/compute/lintstats.go

```diff
@@ -186,24 +96,205 @@ func computeLintStats(lint *rpc.Lint) *rpc.LintStats {
-	for _, pc := range lintstats.ProblemCounts {
-		var problemCount *rpc.LintProblemCount
-		for _, pc2 := range problemCounts {
-			if pc2.RuleId == pc.RuleId {
-				problemCount = pc2
-				break
+func computeLintStatsSpecs(ctx context.Context,
+	client *gapic.RegistryClient,
+	segments []string,
+	filter string,
+	linter string) error {
+
+	return core.ListSpecs(ctx, client, segments, filter, func(spec *rpc.ApiSpec) {
+		// Iterate through a collection of specs and evaluate each.
+		fmt.Printf("%s\n", spec.GetName())
+		// get the lint results
+		request := rpc.GetArtifactContentsRequest{
+			Name: spec.Name + "/artifacts/" + lintRelation(linter) + "/contents",
+		}
+		contents, _ := client.GetArtifactContents(ctx, &request)
+		if contents == nil {
+			return // ignore missing results
+		}
+
+		messageType, err := core.MessageTypeForMimeType(contents.GetContentType())
+		if err != nil {
+			return
+		}
+		if messageType != "google.cloud.apigee.registry.applications.v1alpha1.Lint" {
+			return // ignore unexpected message types
+		}
+
+		lint := &rpc.Lint{}
+		err = proto.Unmarshal(contents.GetData(), lint)
+		if err != nil {
+			return
+		}
+
+		// generate the stats from the result by counting problems
+		lintStats := computeLintStats(lint)
+
+		{
+			// Calculate the operation and schema count
+			request := rpc.GetArtifactContentsRequest{
+				Name: spec.Name + "/artifacts/complexity/contents",
 			}
+			contents, _ := client.GetArtifactContents(ctx, &request)
+			if contents == nil {
+				return // ignore missing results
+			}
+			complexity := &metrics.Complexity{}
+			err := proto.Unmarshal(contents.GetData(), complexity)
+			if err != nil {
+				return
+			}
+
+			lintStats.OperationCount = complexity.GetDeleteCount() +
+				complexity.GetPutCount() + complexity.GetGetCount() + complexity.GetPostCount()
+
+			lintStats.SchemaCount = complexity.GetSchemaCount()
 		}
-		if problemCount == nil {
-			problemCount = &rpc.LintProblemCount{
-				Count:      0,
-				RuleId:     pc.RuleId,
-				RuleDocUri: pc.RuleDocUri,
+
+		storeLintStatsArtifact(ctx, client, spec.GetName(), linter, lintStats)
+	})
+}
+
+func computeLintStatsProjects(ctx context.Context,
+	client *gapic.RegistryClient,
+	segments []string,
+	filter string,
+	linter string) error {
+	return core.ListProjects(ctx, client, segments, filter, func(project *rpc.Project) {
+		if project_segments :=
+			names.ProjectRegexp().FindStringSubmatch(project.GetName()); project_segments != nil {
+			project_stats := &rpc.LintStats{}
+
+			if err := core.ListAPIs(ctx, client, project_segments, filter, func(api *rpc.Api) {
+				aggregateLintStats(ctx, client, api.GetName(), linter, project_stats)
+			}); err != nil {
+				return
 			}
-			problemCounts = append(problemCounts, problemCount)
+			// Store the aggregate stats on this api
+			storeLintStatsArtifact(ctx, client, project.GetName(), linter, project_stats)
```

Comment:
Please change this line to
```
			storeLintStatsArtifact(ctx, client, project.GetName() + "/locations/global", linter, project_stats)
```

This is needed because of the changes in #265 that insert "/locations/global" in resource paths just below projects.